src/core/lib/gprpp/arena.cc

```diff
@@ -77,7 +96,9 @@ void* Arena::AllocZone(size_t size) {
   static constexpr size_t zone_base_size =
       GPR_ROUND_UP_TO_ALIGNMENT_SIZE(sizeof(Zone));
   size_t alloc_size = zone_base_size + size;
-  Zone* z = new (gpr_arena_malloc<GPR_MAX_ALIGNMENT>(alloc_size)) Zone();
+  void* storage = _aligned_alloc<GPR_MAX_ALIGNMENT>(alloc_size);
+  GPR_ASSERT(storage != nullptr);
```

Comment:
Is it actually possible for this to be null?  Linux does allocate-on-write memory management, so most of our code is written to assume that mallocs can never fail.

Reply:
We assume that malloc doesn't fail for the returned pointer because we call abort() if the returned pointer was null - so our code never has to deal with it since we already took the process out to pasture.

This check here is the equivalent of the abort() we do in the existing alloc code.